Provision extra Jitsi JVB services on additional hosts (#2166)
spantaleev/matrix-docker-ansible-deploy: docs/configuring-playbook-jitsi.md

You may want to **limit the maximum video resolution**, to save up resources on both server and clients.

## (Optional) Additional JVBs

> By default a single JVB is deployed on the same hosts as the matrix server. To the scale the JVBs and allow more conferences you may wish to provision additional JVB services on other hosts.

Reviewer: ```suggestion
By default a single JVB ([Jitsi VideoBridge](https://github.com/jitsi/jitsi-videobridge)) is deployed on the same host as the Matrix server. To allow more video-conferences to happen at the same time, you may need to provision additional JVB services on other hosts.
```

There is a ansible role that can be run with the following tag:
` ansible-playbook -i inventory/hosts --limit jitsi_jvb_servers setup.yml --tags=common,setup-additional-jitsi-jvb `

For this role to work you will need an additional section in the ansible hosts file with the details of the JVB hosts, for example:
```
[jitsi_jvb_servers]
<your jvb hosts> ansible_host=<ip address of the jvb host>
```

> Each JVB will require a server id to be set so that it can be uniquely identify and this allows jitsi to keep track of which conferences are on which JVB.  

Reviewer: ```suggestion
Each JVB will require a server id to be set so that it can be uniquely identified and this allows Jitsi to keep track of which conferences are on which JVB.  
```

The server id is set with the variable `matrix_jitsi_jvb_server_id` which ends up as the JVB_WS_SERVER_ID environment variables in the JVB docker container. 

> This variable can be set via the host file, a parameter to the ansible command or in the vars.yaml for the host which will have the additional JVB. For example:

Reviewer: ```suggestion
This variable can be set via the host file, a parameter to the ansible command or in the `vars.yaml` for the host which will have the additional JVB. For example:
```

``` yaml
matrix_jitsi_jvb_server_id: 'jvb-2'
```

``` INI
[jitsi_jvb_servers]
jvb-2.example.com ansible_host=192.168.0.1 matrix_jitsi_jvb_server_id=jvb-2
jvb-3.example.com ansible_host=192.168.0.1 matrix_jitsi_jvb_server_id=jvb-2
```

> Note that the server id jvb-1 is reserved for JVB instance running on the main host and therefore should not be used as the id of an additional jvb host.

Reviewer: ```suggestion
Note that the server id `jvb-1` is reserved for the JVB instance running on the Matrix host and therefore should not be used as the id of an additional jvb host.
```

> The nginx configuration will also need to be updated in order to deal with the additional jvbs server. This is achieved via its own configuration variable

Reviewer: ```suggestion
The nginx configuration will also need to be updated in order to deal with the additional JVB servers. This is achieved via its own configuration variable
```

`matrix_nginx_proxy_proxy_jitsi_additional_jvbs`, which contains a dictionary of server ids to ip addresses.

For example,

``` yaml
matrix_nginx_proxy_proxy_jitsi_additional_jvbs:
   jvb-2: 192.168.0.1
   jvb-3: 192.168.0.2
```

> matrix_nginx_proxy_proxy_jitsi_additional_jvbs: {}

Reviewer: ```suggestion
```

This was probably added here by mistake.

> Applied together this will allow you to provision a extra JVB instances which will register themselves with the prosody service and be available for jicofo 

Reviewer: ```suggestion
Applied together this will allow you to provision extra JVB instances which will register themselves with the prosody service and be available for jicofo 
```
